automation/workflow/hook.go

```diff
@@ -0,0 +1,24 @@
+package workflow
+
+import "github.com/qri-io/qri/event"
+
+// "encoding/json"
+
+// HookType is the type of hook
+type HookType string
+
+// A Hook determines under what circumstances its `hook.Event()` should be
+// emitted, and what the event payload should be.
+type Hook interface {
```

Comment:
we have a notion of `Hook` in the remote package that is a more general form of these: remote hooks can be fired at lots of different procedure lifecycle points. These can only fire when a workflow Succeeds. 

What do you think about calling this interface `CompletionHook` & using this definition:

```go
// CompletionHook is a typed configuration to execute when a workflow transform succeeds
type CompletionHook interface {
```

I'd welcome pushback on this, just trying to capture a succinct, accurate definition.

Reply:
Not necessarily push back, but I don't think we ever fully committed to the idea that hooks only fire when the transform succeeds. Could be that a user wants an alert when it runs, regardless of success.

`Hook` is such a great word, I kept coming back to it even thought I tried to get rid of it when planning. I left off with having the system that listens for these `Hooks` be called `Completors` and the actual emitted "events" be called hooks. I think we eventually want what we call `Hooks` be more what you traditionally think of as a hook and be a place where users can have custom responses to a transform finishing.

Other words I thought could maybe work were `Connectors`, `Linkers`... `DoneTriggers` and then call what we call `Triggers` right now `StartTriggers`... or `KickOffTriggers`

`Kickoffs` and `Connectors`? ... spitballing here

Maybe they should just be called `DoneEvents`? They essentially just describe under what circumstances certain events that signal the end of the workflow be called.

Reply:
hook it is!